imports/lib/ValidateShape.ts

```diff
@@ -0,0 +1,4 @@
+type ValidateShape<T, Shape> =
+  Shape & { [K in keyof T]: K extends keyof Shape ? T[K] : never };
```

Comment:
This is much simpler than the old declaration of `ValidateShape`, but I think should have the same effect (and I believe that the new test confirms that)